pinot-core/src/main/java/org/apache/pinot/core/util/SchemaUtils.java

```diff
@@ -92,6 +93,8 @@ public static void validate(Schema schema) {
                 "Exception in getting arguments for transform function '" + transformFunction + "' for column '"
                     + column + "'", e);
           }
+        } else {
+          primaryKeyColumnCandidates.add(column);
```

Comment:
Any specific reason why transformed column cannot be primary key column?

Reply:
we can remove this constraint, though I feel it shall be very rare for the transformed column to be the primary key